client/rest-high-level/src/test/java/org/elasticsearch/client/ESRestHighLevelClientTestCase.java

```diff
@@ -131,7 +134,12 @@ protected static RestHighLevelClient highLevelClient() {
 
     private static class HighLevelClient extends RestHighLevelClient {
         private HighLevelClient(RestClient restClient) {
-            super(restClient, (client) -> {}, new SearchModule(Settings.EMPTY, false, Collections.emptyList()).getNamedXContents());
+            super(
+                restClient,
+                (client) -> {},
+                Stream.concat(
+                    new SearchModule(Settings.EMPTY, false, Collections.emptyList()).getNamedXContents().stream(),
+                    new MlDataFrameAnalysisNamedXContentProvider().getNamedXContentParsers().stream()).collect(toList()));
```

Comment:
It's a bit untidy that this has to be done here. `EsIntegTestCase` has the more extensible approach of using a virtual method to gather the NamedXContentRegistry entries. What do yo think of doing something similar?

```
    @Override
    protected NamedXContentRegistry xContentRegistry() 
```

Reply:
Virtual method approach doesn't work for integration test classes (I don't why).
But I was able to solve this by adding my XContentProvider to client/rest-high-level/src/main/resources/META-INF/services/org.elasticsearch.plugins.spi.NamedXContentProvider file.

Reply:
Neat I hadn't thought of that.